### PR TITLE
#208 split PR 3, rename savedata to saveData

### PR DESCRIPTION
**Pull request by red031000**
camelCase good
also renames a few save funcs to be consistent with the style

### CODE REVIEW

**Comment by luckytyphlosion** [comment]
## Renamed Functions
`GetStoragePCPointer` -> `Save_PCStorage_Get`
`FieldSystem_GetSaveDataPtr` -> `FieldSystem_GetSaveData`
`SavGymmick_GetType` -> `Save_Gymmick_GetType`
`FieldSystem_IsSavGymmickTypeEqualTo` -> `FieldSystem_IsSaveGymmickTypeEqualTo`
`SavGymmick_AssertMagic_GetData` -> `Save_Gymmick_AssertMagic_GetData`
`SavGymmick_Init` -> `Save_Gymmick_Init`

## Renamed Structs and Fields
`savedata` -> `saveData`

File: src/field_system.c [hunk 1]
```diff
@@ -307,8 +307,8 @@ BgConfig *FieldSystem_GetBgConfigPtr(FieldSystem *fieldSystem) {
     return fieldSystem->bgConfig;
 }
 
-SaveData *FieldSystem_GetSaveDataPtr(FieldSystem *fieldSystem) {
-    return fieldSystem->savedata;
+SaveData *FieldSystem_GetSaveData(FieldSystem *fieldSystem) {
```
Comment (luckytyphlosion) [inline src/field_system.c:310]: I wonder if it's a good idea to have `Ptr` as implicit when other `Get` functions may return a struct and not a pointer.

Comment (red031000) [reply src/field_system.c:310]: I litearlly know zero get functions that return a struct and not a pointer to one

File: src/save_arrays.c [hunk 2]
```diff
@@ -267,7 +267,7 @@ struct UnkStruct_0202E474 *sub_020270C4(SaveData *saveData) {
     return SaveArray_Get(saveData, SAVE_UNK_23);
 }
 
-PC_STORAGE *GetStoragePCPointer(SaveData *saveData) {
+PC_STORAGE *Save_PCStorage_Get(SaveData *saveData) {
```
Comment (luckytyphlosion) [inline src/save_arrays.c:270]: Few things:
1. Why `Save_PCStorage_Get` and not `Save_GetPCStorage`?
2. Shouldn't this be `SaveArray_PCStorage_Get`/`SaveArray_GetPCStorage`
3. `SaveArray_Get` is probably a bad name in itself, it's more like `SaveArray_GetEntry` (`SaveArray_Get` implies getting the entire SaveArray itself)

File: src/save_gymmick.c [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
 }
 
-union GymmickUnion *SavGymmick_AssertMagic_GetData(struct Gymmick *gymmick, int kind) {
+union GymmickUnion *Save_Gymmick_AssertMagic_GetData(struct Gymmick *gymmick, int kind) {
     GF_ASSERT(gymmick->kind == kind);
     return &gymmick->data;
 }
 
-int SavGymmick_GetType(struct Gymmick *gymmick) {
+int Save_Gymmick_GetType(struct Gymmick *gymmick) {
```
Comment (luckytyphlosion) [inline src/save_gymmick.c:19]: What does `Save` refer to here exactly? Why is this file called `save_gymmick.c`?

Comment (luckytyphlosion) [reply src/save_gymmick.c:19]: I feel like this isn't `Save` at all and it's just `Gymmick`, `Save` usually means that the functions actually directly access the Save Array but none of the functions here do.

File: src/save_arrays.c (hunk 2 above)
Comment (red031000) [reply src/save_arrays.c:270]: 1. consistent with all the other getters
2. probably
3. idk

File: src/save_gymmick.c (hunk 3 above)
Comment (red031000) [reply src/save_gymmick.c:19]: out of scope

Comment (adrienntindall) [reply src/save_gymmick.c:19]: this is a save PR it isnt outta scope

Comment (red031000) [reply src/save_gymmick.c:19]: this pr is renaming savedata to saveData, classifying whether Gymmick belongs under Save is out of scope (basically I'd prefer to do this in another PR)

**Review by adrienntindall — APPROVED** [review]

**Pull request merged**